### ISSUE TITLE
diff file and possible bug using multiple asterisks

### ISSUE THREAD

**Issue by arturmeerzon** [highlightjs/highlight.js#2038]
Hello everyone,

i tried highlightjs v9.15.6 for displaying diff files and I noticed some strange kind of behavior when multiple asterisks are used, all lines after the line with asterisks are just ignored, the example is below:

```
+line
 5 or more ***** asterisks between + or - are causing bug

+this line is ignored
+this one too
+etc
```

I wrap the code as you mentioned in docs:
`<pre><code class="diff">...</code></pre>`

I haven't tried to test it with other languages yet.

**Comment by egor-rogov** [comment]
Hi,
it comes that our implementation of diff explicitly considers 5 or more asterisks as a comment (until the end of the line). I believe it was done with some thought, although I am not sure what was the reason.

**Comment by arturmeerzon** [comment]
yes removing this line would fix this

```
{begin: /\*{5}/, end: /\*{5}$/}
```

just wondering what where the reasons

**Comment by yyyc514** [comment]
@egor-rogov Looks like a mistake to me.  This was done in the middle of a huge change to file formats and has no other back story or reasoning... it wasn't in the original... seems it's just a mistake to me.

https://github.com/highlightjs/highlight.js/commit/c48cff9d445f162ea940067f5e2302a557ea39e1#diff-8d12b71f145c127832d1f1feebd7f829R27

I've read a LOT of diffs and never seen anything like a ***** comment.

**Comment by yyyc514** [comment]
Oh wait, I see it now, let me follow it further back.

**Comment by yyyc514** [comment]
Ok, it goes all the way back to forever:

https://github.com/highlightjs/highlight.js/commit/21c0aa3ac4b3f9c52fff379d821334eaeabb85b0#diff-1f15fcf0ece01f7fc81ba10d8a695fddR61

**Comment by yyyc514** [comment]
Ok, god I love open source.  This is context format. And it's supposed to be 15 of them at the BEGINNING of a line: 

https://www.gnu.org/software/diffutils/manual/html_node/Example-Context.html#Example-Context

I check the diffutils source also.

**Comment by yyyc514** [comment]
https://github.com/highlightjs/highlight.js/pull/2159

**Comment by egor-rogov** [comment]
Fixed in #2159
